crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -499,6 +524,103 @@ impl<'a> TimelineEventHandler<'a> {
         self.state.reactions.map.insert(reaction_id, (reaction_sender_data, c.relates_to));
     }
 
+    #[instrument(skip_all, fields(replacement_event_id = ?replacement.event_id))]
+    fn handle_poll_start_edit(
+        &mut self,
+        replacement: Replacement<UnstablePollStartEventContentWithoutRelation>,
+    ) {
+        update_timeline_item!(self, &replacement.event_id, "poll edit", |event_item| {
+            if self.ctx.sender != event_item.sender() {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Edit event applies to another user's timeline item, discarding"
+                );
+                return None;
+            }
+
+            let poll_state = if let TimelineItemContent::Poll(poll_state) = &event_item.content() {
+                poll_state
+            } else {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Can't edit a poll that is not of type TimelineItemContent::Poll, discarding"
+                );
+                return None;
+            };
+
+            let new_content =
+                if let Ok(edited_poll_state) = poll_state.edit(&replacement.new_content) {
+                    TimelineItemContent::Poll(edited_poll_state)
+                } else {
+                    info!(
+                        original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                        "This poll can't be edited, discarding"
+                    );
+                    return None;
+                };
+
+            let edit_json = match &self.ctx.flow {
+                Flow::Local { .. } => None,
+                Flow::Remote { raw_event, .. } => Some(raw_event.clone()),
+            };
+
+            trace!("Applying edit");
+            Some(event_item.with_content(new_content, edit_json))
+        });
+    }
+
+    fn handle_poll_start(&mut self, c: UnstablePollStartEventContent, should_add: bool) {
+        let mut poll_state = PollState::new(c);
+        // TODO: If the flow is not remote we're not able to apply the pending events. Is this a problem?
+        if let Flow::Remote { event_id, .. } = self.ctx.flow.clone() {
+            self.state.poll_pending_events.apply(&event_id, &mut poll_state);
+        };
```

Comment:
```suggestion
        }
```

Reply:
what's the difference?

Reply:
None. I just find redundant semicolons after closing braces a bit confusing / weird.